src/ufuncs.c

```diff
@@ -984,6 +875,68 @@ static void YY_Y_func(char** args, npy_intp* dimensions, npy_intp* steps, void*
 }
 static PyUFuncGenericFunction YY_Y_funcs[1] = {&YY_Y_func};
 
+/* Define the reducing geoms -> geom functions (Y_Y_reduce) */
+static void* intersection_all_data[1] = {GEOSIntersection_r};
+static void* symmetric_difference_all_data[1] = {GEOSSymDifference_r};
+static char Y_Y_reduce_dtypes[2] = {NPY_OBJECT, NPY_OBJECT};
+static void Y_Y_reduce_func(char** args, npy_intp* dimensions, npy_intp* steps,
+                                   void* data) {
+  FuncGEOS_YY_Y* func = (FuncGEOS_YY_Y*)data;
+  GEOSGeometry* geom = NULL;
+  GEOSGeometry* ret_ptr = NULL;
+  int n_geoms;
+
+  CHECK_NO_INPLACE_OUTPUT(1);
+
+  GEOS_INIT;
```

Comment:
I started from the `polygonize` implementation, which is using OUTPUT_Y and thus not releasing the GIL. But we can use the `geom_arr` temporary array to store output GEOSGeometries trick here as well to be able to release the GIL.